doc/tracing.md

```diff
@@ -0,0 +1,201 @@
+Tracing
+=======
+
+To make it possible to have more detailed logging for troubleshooting, or for keeping track of detailed statistics, Bitcoin Core makes use of so-called eBPF tracepoints.
+
+The linux kernel can hook into these tracepoints at runtime, but when disabled they have little to no performance impact. Traces can pass data which can be printed externally via tools such as `bpftrace`.
+
+Viewing all tracepoints
+-----------------------
+
+To view all available probes in a binary, use `info probes` in `gdb`:
+
+```
+$ gdb src/bitcoind
+…
+(gdb) info probes
+Type Provider   Name            Where              Semaphore Object                                     
+stap net        process_message 0x00000000000d48f1           /…/bitcoin/src/bitcoind 
+stap net        push_message    0x000000000009db8b           /…/bitcoin/src/bitcoind 
+stap validation connect_block   0x00000000002adbe6           /…/bitcoin/src/bitcoind
+…
+```
+
+Tracepoint documentation
+------------------------
+
+The current tracepoints are listed here.
+
+### net
+
+- `net:process_message(char* addr_str, int node_id, char* msgtype_str, uint8_t* rx_data, size_t rx_size)`
```

Comment:
I would prefer to rearrange the arguments to `push_message` to be in the same order as process_message (eg move node_id up front):
```
net:push_message(int node_id, char* msgtype_str, uint8_t* tx_data, size_t tx_size)
```